lib/node/rpc.js

```diff
@@ -1960,7 +1960,6 @@ class RPC extends RPCBase {
             key.script = redeem;
             key.witness = script.isWitnessScripthash();
             key.refresh();
-            break;
```

Comment:
you might want to use this if you want to apply a different SIGHASH flag per input. (currently there seems to be only one SIGHASH type supported being applied to all inputs signatures per signature request, per input).

If you do want different SIGHASH types per input then you'll need this `break` in place  and you'll need to resend the tx to be signed as many times as the nr. of inputs to be signed.

If we remove the break line, we can atomically sign multiple inputs with the same SIGASH type, but we loose the flexibility of allowing multiple

Reply:
Looking at Bitcoin Core `signrawtransaction`, only one SIGHASH parameter is accepted by the RPC call which, I assume, applies to all inputs in the tx. In our function, SIGHASH is handled here: https://github.com/bcoin-org/bcoin/blob/master/lib/node/rpc.js#L1969-L1986 So I think that we are OK for those.

However you are right that removing the `break` means a private key can only be used with one `script`and `witness` type in a transaction because of the `map`.  I'm not sure how Bitcoin Core would handle that case but here it would fail if the same private key was used differently in two different inputs of the same transaction:

**Failure case with different `script`'s:**
input 0: Multisig `AlicePrivate` + `BobPrivate`
input 1: Multisig `AlicePrivate` + `CharliePrivate`

**Failure case with different `witness`'s:**
input 0: P2SH `AlicePrivate` (legacy BIP16 script hash, multisig or not)
input 1: P2WSH `AlicePrivate` (SegWit BIP141 script hash, mutisig or not)